### ISSUE TITLE
easier lookup for local identifiers

### ISSUE THREAD

**Issue by JJ-Author** [dbpedia/dbp-same-thing-service#10]
I was approached whether we could add a "namespace" parameter to deliver only the local identifiers of the namespace / datasource of interest

example usage scenario:
get LOCAL wikidata identifier(s) for one DBpedia Identifier
localsNamespace="wikidata.org"

**Comment by JJ-Author** [comment]
@aolieman would this decrease performance because local identiifers would need to be filtered out in the response or could this potentially speed up the request?

**Comment by aolieman** [comment]
This seems easy enough. I expect it has low impact on performance: a couple of extra cpu cycles to filter local urls. If we can limit the filtering to match namespaces only at the start of local IRIs (e.g. `/^www\.wikidata\.org/`, this should have a negligible effect on load & latency. I think @kurzum already requested the option to make IRI matching insensitive to the protocol (http/https/...), so we could make filtering on namespace efficient if we disregard protocol.

I'd probably make the implementation of a namespace filter similar to the client-side implementation in nifconverter: https://github.com/wetneb/nifconverter/blob/master/nifconverter/dbpedia_samething.py

**Comment by JJ-Author** [comment]
yeah the protocol has a tiny problem we can not expect from every namespace that an http vs. https IRI represents the same thing. for dbpedia we decided that this is the same though.

**Comment by aolieman** [comment]
That would only be a problem for namespace filtering if these IRIs that only differ in protocol, but represent different things would both be loaded in the STS.

For example, suppose the following exist and don't represent the same:
- http://my.namespace.dev/entity/
- https://my.namespace.dev/entity/

If the STS only contains https://my.namespace.dev/entity/, but the user filters on namespace http://my.namespace.dev/, both lookup keys would internally be shortened to start with "my.namespace.dev/" and the matching entity IRI will be returned (its full form, starting with protocol). Perhaps the user does not want this weak match, but if the user wanted to only match http://my.namespace.dev/, they could do stricter filtering client-side.

I'd have to see an example with real IRIs where this is a problem to reconsider the approach. If that poses a legitimate issue, I would propose to not make IRI matching insensitive to the protocol at all.

**Comment by kurzum** [comment]
@aolieman @JJ-Author  there should be no difference between http/s identifiers. Linked data is more liberal than RDF, but there shouldn't be a case where it's different.

Actually I was hoping that w could normalize input with some rules, e.g. https://dbpedia.org.   Can we check the logs and then write 5 or 10 so we get better hits?

The filter request is annoying as they want us to implement sth. that can easily be done clientside. This one adds a bit of convenience, so it's OK, it would be OK to decline it as well.

Regarding the filter, it should either take Regex or fully qualified domain names or just domain, e.g. namespace=it.dbpedia.org

**Comment by aolieman** [comment]
I'd go for the "just domain" filter option as it's easy to implement and use. Regex would also be fine.

I don't understand this question:
> Actually I was hoping that w could normalize input with some rules, e.g. https://dbpedia.org. Can we check the logs and then write 5 or 10 so we get better hits?

**Comment by kurzum** [comment]
@aolieman not sure if domain is the easiest. I think fully qualified domain is the easiest, i.e. including www or it.dbpedia.org

at the moment, if somebody asks for http*s*://dbpedia.org/resource/Berlin he will get an error, but he could as well get a redirect to http://dbpedia.org/resource/Berlin 

It would fix input heterogenity on the part before the `/` . For example, I always forget what the correct first part for Wikidata is. They have a confusing setup.
